packages/core/components/FileMetadataSearchBar/index.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+    DatePicker,
+    Dropdown,
+    Icon,
+    IconButton,
+    IDropdownOption,
+    SearchBox,
+} from "@fluentui/react";
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { AnnotationName, TOP_LEVEL_FILE_ANNOTATIONS } from "../../constants";
+import { AnnotationType } from "../../entity/AnnotationFormatter";
+import FileFilter from "../../entity/FileFilter";
+import { selection } from "../../state";
+import { getFileAttributeFilter } from "../../state/selection/selectors";
+
+const styles = require("./FileMetadataSearchBar.module.css");
+
+const FILE_ATTRIBUTE_OPTIONS = TOP_LEVEL_FILE_ANNOTATIONS.filter(
+    (a) => a.name !== AnnotationName.FILE_SIZE
+).map((a) => ({
+    key: a.name,
+    text: a.displayName,
+}));
+const FILE_NAME_OPTION = FILE_ATTRIBUTE_OPTIONS.find(
+    (o) => o.key === AnnotationName.FILE_NAME
+) as IDropdownOption;
+export const DATE_RANGE_SEPARATOR = "-to-"; // Not arbitrary, defined per contract with FES
+
+// Because the datestring comes in as an ISO formatted date like 01-02-21
+// creating a new date from that would result in a date displayed as the
+// day before due to the UTC offset, to account for this we can add in the offset
+// ahead of time.
+function extractDateFromDateString(dateString?: string): Date | undefined {
+    if (!dateString) {
+        return undefined;
+    }
+    const date = new Date(dateString);
+    date.setMinutes(date.getTimezoneOffset());
+    return date;
+}
+
+/**
+ * This component renders a dynamic search bar for querying file records by
+ * basic file attributes that are otherwise not queryable through usual means like
+ * in the <AnnotationSidebar /> for example.
+ */
+export default function FileMetadataSearchBar() {
+    const dispatch = useDispatch();
+    const fileAttributeFilter = useSelector(getFileAttributeFilter);
+    const [selectedAttribute, setSelectedAttribute] = React.useState<IDropdownOption>(
+        FILE_NAME_OPTION
+    );
+
+    function onResetSearch() {
+        if (fileAttributeFilter) {
+            dispatch(selection.actions.removeFileFilter(fileAttributeFilter));
+        }
+    }
+
+    function onSearch(filterValue: string) {
+        if (filterValue && filterValue.trim()) {
+            const fileFilter = new FileFilter(selectedAttribute.key as string, filterValue);
+            if (fileAttributeFilter) {
```

Comment:
Modify this logic a bit and wrap the actions to be dispatched in a `batchActions`?

Reply:
They have middleware that would otherwise get skipped in batchActions (unless that doesn't happen in this repo?)

Reply:
Ah, that sounds correct